chore(deps): upgrade dependencies to the latest

Jexsie opened the pull request:
Continuation for https://github.com/OpenDataEnsemble/ode/pull/68 since it was merging to the main

Bahati308 committed:
Add GitHub issue templates

- Add bug report template with component selection
- Add feature request template
- Configure to prevent blank issues
- Add links to documentation and discussions

Bahati308 committed:
feat: enhance bug report template with user story and acceptance criteria

Add optional User Story, Details, and Acceptance Criteria sections to
improve bug reproduction and testing capabilities.

Jexsie committed:
- use ES module syntax
- upgrade packages to the latest
- add a pnpm lock file
- update config files to use the es module syntax
Signed-off-by: Jessie Ssebuliba <jessiessebuliba@gmail.com>

Signed-off-by: Jessie Ssebuliba <jessiessebuliba@gmail.com>

Jexsie commented on formulus/eslint.config.js:
```diff
@@ -0,0 +1,52 @@
+      ...hooksPlugin.configs.recommended.rules,
+      'react/react-in-jsx-scope': 'off',
+      'react-native/no-inline-styles': 'warn',
+      'react-native/no-color-literals': 'warn',
```
This is to enforce @IamLRB's UI tokens. All occurances of iniline styles and color literals will be trigger warnings. 
All color literals will have to be defined in the tokens

Jexsie committed:
use a central theme provided by @najuna-brian for consistent theming, later has to be migrated to the UI tokens after review

Signed-off-by: Jessie Ssebuliba <jessiessebuliba@gmail.com>

Jexsie committed:
use node 24 instead 25

Signed-off-by: Jessie Ssebuliba <jessiessebuliba@gmail.com>

Jexsie committed:
update all config files use es module syntax

Signed-off-by: Jessie Ssebuliba <jessiessebuliba@gmail.com>

Jexsie committed:
chore! fix eslint breaking issues

Signed-off-by: Jessie Ssebuliba <jessiessebuliba@gmail.com>

Jexsie committed:
more linting changes

Signed-off-by: Jessie Ssebuliba <jessiessebuliba@gmail.com>

Jexsie committed:
ignore linting of all webview scripts

Signed-off-by: Jessie Ssebuliba <jessiessebuliba@gmail.com>

Jexsie committed:
add prettier rewrite changes

Signed-off-by: Jessie Ssebuliba <jessiessebuliba@gmail.com>

Jexsie committed:
fix(android): enable new architecture to resolve native codegen build failures

Signed-off-by: Jessie Ssebuliba <jessiessebuliba@gmail.com>

Jexsie committed:
fix(ui): update EmptyState icon typing for RN Vector Icons v12+

- Adapted EmptyState.tsx to use type-safe icon names
- Replaced generic string with keyof MaterialIcons.glyphMap
- Ensures TypeScript validation with react-native-vector-icons v12+

Signed-off-by: Jessie Ssebuliba <jessiessebuliba@gmail.com>

Jexsie committed:
Add pretty changes

Signed-off-by: Jessie Ssebuliba <jessiessebuliba@gmail.com>

Jexsie committed:
Rebase and lint

Signed-off-by: Jessie Ssebuliba <jessiessebuliba@gmail.com>

Jexsie committed:
Ignore test for Q2

Signed-off-by: Jessie Ssebuliba <jessiessebuliba@gmail.com>

Jexsie commented on formulus/android/app/src/main/java/org/opendataensemble/formulus/MainApplication.kt:
```diff
@@ -5,45 +5,25 @@ import android.webkit.WebView
+        add(UserAppPackage())
+      },
+    )
+  }
```
These changes officially retires the legacy ReactNativeHost (The Bridge) in favor of the new ReactHost

**Key Changes:**
Bridgeless by Default: Enabled the modern Bridgeless runtime using getDefaultReactHost.

Switched to loadReactNative(this), which simplifies the initialization of native components and architecture flags.

Re-registered UserAppPackage within the new reactHost lazy initializer to ensure compatibility via the New Architecture Interop Layer.

r0ssing commented:
Thanks @Jexsie ! Let's either get this PR back on track and merge it quickly or split it up into more manageable bits 😺 
Will this somehow also make the renovate-PR's redundant?
@Ndacyayisenga-droid will you have a look? 👀

Jexsie committed:
fix(ios): Fix webview asset loading by bundling assets and using MainBundlePath

- Add webview assets to iOS bundle via react-native-asset
- Update placeholder and injection script paths to use RNFS.MainBundlePath
- Update .gitignore to exclude xcuserdata directories

Signed-off-by: Jessie Ssebuliba <jessiessebuliba@gmail.com>

Jexsie committed:
fix lint errors

Signed-off-by: Jessie Ssebuliba <jessiessebuliba@gmail.com>

Jexsie committed:
fix(ios): improve asset loading and console logging for iOS platform

- Fix iOS webview asset loading by setting allowingReadAccessToURL to MainBundlePath for bundled assets
- Add formplayer_dist folder reference to Xcode project for proper bundled asset access
- Update copy-to-rn.js script to copy formplayer build to both Android and iOS directories
- Improve console log transport with proper Error serialization and circular reference handling
- Enhance console message forwarding with better error handling and proper binding
- Update clean-rn-assets.js to support cleaning both Android and iOS asset directories
- Add formplayer_dist to gitignore to prevent tracking generated assets

This fixes the issue where iOS was unable to load webview assets properly due to
incorrect file access permissions and missing folder references in the Xcode project.

Signed-off-by: Jessie Ssebuliba <jessiessebuliba@gmail.com>

Jexsie committed:
run lint and pretty

Signed-off-by: Jessie Ssebuliba <jessiessebuliba@gmail.com>

Jexsie commented:
<img width="456" height="839" alt="Screenshot 2026-01-26 at 18 06 50" src="https://github.com/user-attachments/assets/5ba39721-2986-4de2-9d1d-419726917eb8" />
<img width="456" height="839" alt="Screenshot 2026-01-26 at 18 07 52" src="https://github.com/user-attachments/assets/243b8bdc-89f6-4f0d-a982-65a50caea15f" />

Jexsie committed:
chore: upgrade tooling and standardize configs for formplayer and portal

- Add Prettier configs matching formulus standards
- Upgrade to ESLint 9 with flat config
- Update TypeScript and dependencies to latest
- Add format scripts and improve lint tooling
- Exclude React Native configs (web-only projects)

Signed-off-by: Jessie Ssebuliba <jessiessebuliba@gmail.com>

Jexsie committed:
feat(formplayer,portal)!: migrate from react-scripts to Vite

Replace deprecated Create React App with Vite 6.4 for 10x faster builds
across formulus-formplayer and portal applications.

BREAKING CHANGES:
- Build system: webpack → Vite
- ESLint: old format → flat config
- Environment variables: process.env → import.meta.env

Changes (both projects):
- Add vite@^6.4.1, @vitejs/plugin-react@^4.7.0
- Remove react-scripts, react-app-rewired
- Create vite.config.ts with optimized code splitting
- Move index.html to root, remove %PUBLIC_URL% placeholders
- Update scripts: start (vite), build (tsc && vite build), test (vitest)
- Migrate ESLint to flat config (eslint.config.js)
- Update tsconfig.json target from es5 to ES2020
- Delete config-overrides.js (replaced by Vite config)

Formplayer-specific:
- Fix dev dependency imports to use synchronous ES modules
- Add vite/client types reference for import.meta.env support
- Update webview-mock initialization to work with Vite

Performance improvements:
- Dev server startup: ~10-20s → <1s ⚡
- Hot module replacement: 1-3s → ~100ms ⚡⚡⚡
- Production build: ~60s → ~15-20s ⚡⚡

Signed-off-by: Jessie Ssebuliba <jessiessebuliba@gmail.com>

Jexsie committed:
(chore) use stable node version instead of latest

Signed-off-by: Jessie Ssebuliba <jessiessebuliba@gmail.com>

Ndacyayisenga-droid commented:
@Jexsie Any updates on this?

Jexsie committed:
add prettier rewrite changes

Signed-off-by: Jessie Ssebuliba <jessiessebuliba@gmail.com>

Jexsie committed:
fix lint errors

Signed-off-by: Jessie Ssebuliba <jessiessebuliba@gmail.com>

Jexsie committed:
fix lint errors

Signed-off-by: Jessie Ssebuliba <jessiessebuliba@gmail.com>

Jexsie committed:
ignore formulus tests

Signed-off-by: Jessie Ssebuliba <jessiessebuliba@gmail.com>

Ndacyayisenga-droid commented:
@Jexsie could you resolve the conflicts?

Jexsie committed:
rebase

Signed-off-by: Jessie Ssebuliba <jessiessebuliba@gmail.com>

Jexsie committed:
fix up

Signed-off-by: Jessie Ssebuliba <jessiessebuliba@gmail.com>

Jexsie committed:
use es6 syntax for scripts in the formplayer

Signed-off-by: Jessie Ssebuliba <jessiessebuliba@gmail.com>
